perf/ring.go

```diff
@@ -148,13 +167,37 @@ func (rr *ringReader) Read(p []byte) (int, error) {
 	if remainder := int(rr.head - rr.tail); n > remainder {
 		n = remainder
 	}
+	end = start + n
+	return
+}
 
-	copy(p, rr.ring[start:start+n])
+func (rr *ringReader) readUint32() (uint32, error) {
+	n := 4
+	start, end := rr.getBounds(n)
+	if end-start < n {
+		return 0, io.EOF
```

Comment:
This doesn't handle the case where a number is split across the beginning / end of the ring buffer?

Reply:
Good call! Let me work on that.

Reply:
Just to make sure, you saw https://github.com/cilium/ebpf/pull/617#pullrequestreview-928250434 ?